### PR TITLE
update release workflow to tag images with canary(-nonroot) instead of latest(-nonroot)

### PR DESCRIPTION
**Pull request by tpetr**
test plan
- checks pass
- release dry run looks good

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -92,9 +92,16 @@ jobs:
...
+      #
+      # We first push to canary and then promote to latest
+      # once we confirm that the release doesn't have any
+      # obvious bugs or issues
+      #
       docker-flavor: |
-        latest=auto
+        latest=false
       docker-tags: |
+        type=raw,value=canary
```
Comment (aryx) [inline .github/workflows/release.yml:100]: would be good to explain those 3 lines

**Review by aryx — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -93,9 +93,14 @@ jobs:
     needs: [inputs]
     with:
       docker-flavor: |
-        latest=auto
+        # don't add a "latest" tag (we'll promote "canary-nonroot" to "latest-nonroot" after testing)
```
Comment (aryx) [inline .github/workflows/release.yml:96]: I think here it's just canary to latest (there is no "nonroot" suffix).

**Comment by aryx** [comment]
thx for the comments! I now understand those lines.

File: .github/workflows/release.yml (hunk 2 above)
Comment (tpetr) [reply .github/workflows/release.yml:96]: good catch, copypasta fail

**Pull request merged**